### ISSUE TITLE
Question: How can I Sponsor getting things done?

### ISSUE THREAD

**Issue by mrpastewart** [AvaloniaUI/Avalonia#1151]
I have been following this project for sometime and in my opinion it needs a push. I say this because even in alpha you have managed to do what everybody is asking Microsoft to do and that is get a XAML UI framework running on dotnetcore (netstandard). There are others but none have managed to get on the new core or standards as of yet. I want to use this and depend on it's success so I want to know if we can put a system in play to sponsor bug fixes and enhancements. I could maybe hire a developer to work dedicated on fixing issues but there is a community forming here so I figure why not just put up a bounty so we can clear the backlog and add some of the enhancements I see in the issues.

I have much respect for what is going on here and I feel it has the potential to become the defacto cross platform UI for all .Net Platforms. Maybe even garner some official love from the powers above. In some ways it reminds me of a baby opensource managed version of noesis I just need to create applications that run where I need them to and I am willing to bounty up and support this project.

Bottom line is you never know what may happen in the OSS world. Look at The man behind mono he pushed and advocated and pushed and hit paydirt when Microsoft finally listened. No one not even MS themselves have managed to address what this project has put in motion today (just like the current state of .NET itself has greatly benefited from the knowledge of the mono team who did it all first xplatform, mobile all that MS would be nowhere without that knowledge). You may be sitting on a goldmine if we can get it to Beta, RC and production and in real world LOB, IOT and general platform application development others who are not paying attention just may do so.

**Comment by kekekeks** [comment]
I guess we could use https://www.bountysource.com/

However, there are several concerns about money being involved in OSS summarized by Sean Owen there: https://www.quora.com/Why-doesnt-Github-add-a-bounty-feature-so-that-people-could-conveniently-pay-for-issues-to-get-fixed/answer/Sean-Owen

So we'll need a system to prevent to gross hacks being pushed to upstream.

**Comment by VladimirAkopyan** [comment]
I think this man has his the head on the nail. 
This project addresses a massive need, in a way nothing else really does. Given that current project leads have accomplished much, we'd be happy to trust them with a bit of cash.

**Comment by mrpastewart** [comment]
I think with a good system in place putting some monetary incentives in the mix can be a good thing. For example. 

1. Issues (bugs, task) get filed, reproduced and verified that are directly attached to a release or milestone.
2. Issues get filed that are feature request and/or enhancements.
3. The project lead can then pursue dialog with the community using the issues comment sections to get a pulse from the community about feature request or enhancements ultimately allowing the community to chime in on issues.
4. once this happens votes can be taken if a feature/enhancement will make it into the project and that it fits the scope and goals of where Avalonia is going.
5. Obviously bugs and task such as refactoring and general project maintenance will need to be reproduced and verified then assigned to be fixed.
6. Using bountysource the lead (or his designee) can now push up for grab issues or anything that the leads need help getting done to the site and sponsor bounties as incentives to make it happen.

We don't want any gross hacks, We don't want any deviations from scope and goals. All code must pass unit test and follow code of conduct rules for the project. What we do want is for the repo to always have activity moving forward as fast as the community dictates. Shy of having dedicated developers working on the project full time this may be the only viable way to get the project lead some help outside those in the community already pushing to the mainline.

What the goal should be is to get a production ready release and then maintain it from there using all means at our disposal. Checks and Balances are needed for sure but this project has great, great potential and I want it LOL.

I'm just thinking out load here correct me if I'm out of line or where I may be off.

**Comment by grokys** [comment]
Good question @mrpastewart! I've had less time (or at least less consistent time) to work on Avalonia recently, but it's been great seeing the community step up and fill the gap. If we could use sponsorship to accelerate this process it would be a win/win for everyone I think.

I think @kekekeks concerns are very much valid but I think your thoughts outlined above are a good start. However we do it though, I think we should be aiming to get longer-term contributors and maintainers onboard rather than parachuting people in for a single feature.

Thanks for starting this discussion @mrpastewart!

**Comment by mrpastewart** [comment]
I Agree totally that we need long term contributors and maintainers which I think will happen naturally once the project reaches production status and starts showing up in LOB applications. By default use = exposure = interest = participation just my view.

**Comment by grokys** [comment]
Any idea of how we could get this started? Perhaps a trail run of some description? Are there any features in particular that you had in mind?

**Comment by mrpastewart** [comment]
Yes my goal is to build a cross-platform Transportation Management System using Avalonia. For this it would require addition of some new controls such as a datagrid. These issues come to mind but also we should initially include all verified bugs and get them out of the backlog

https://github.com/AvaloniaUI/Avalonia/issues/1030
https://github.com/AvaloniaUI/Avalonia/issues/1041
https://github.com/AvaloniaUI/Avalonia/issues/1002
https://github.com/AvaloniaUI/Avalonia/issues/1134

**Comment by kekekeks** [comment]
I guess, `Calendar` and `StatusBar` are small enough to be funded via something like bountysource.

The problem about DataGrid is that it's a really, really huge task. WPF didn't have one until .NET 4.0. Even if we port one from Xceed, it's still a ~130K LOC beast. And we probably lack some subsystems to get it running properly. It might require a full time developer to be working on it for several months. So that task just can't be accomplished without proper funding.

**Comment by mrpastewart** [comment]
BountySource allows for more then just bounties on issues for bigger projects we can use the salt feature which is actually designed to allow developers to earn monthly income for their work hence we can actually pay a full time developer thru this platform check it out https://salt.bountysource.com/ to understand more about it but in a nutshell Salt is a crowdfunding platform that helps developers earn a monthly salary from open-source software. So we can still use bountysource for bigger stuff that may require a dedicated dev.

**Comment by mindyourservice** [comment]
Hi Guys,

Great project BTW!!! I think Mr Stewart is correct when he says there are many people who would love to see this go commercial and be used as the LOB app go to.

Telerik controls are a must for us... But perhaps their UWP controls are enough as they include datagrids and are open source.

Am I too naive to assume that these could be ported to Avalonia. It would give you a nice set of base controls that are commercial grade and possibly inspire Telerik to port their SL control set.

**Comment by mrpastewart** [comment]
@mindyourservice  I am not actually advocating commercializing this project although I strongly believe it is possible hence the "your sitting on a goldmine" comment above. What I am advocating right now is a way to crowdfund bug fixes, enhancements and feature request. I do believe this project will in the future have commercial capacity. I won't lie and say in the end that I don't hope to see that happen as we would have a much stronger product and support channel if it paid someone's bills. I would pay in a drop of a bucket but for now I will help pay to get it there.

**Comment by kekekeks** [comment]
> where one of the vendors like Telerik start making controls that work with Avalonia

I had a talk with a representative of one of those control-making companies (can't disclosure which one) a year ago on a conference. They were interested, but the quality of the project at that time wasn't quite satisfactory. I'll contact that person again when we'll release beta and see if it leads to something.

**Comment by mindyourservice** [comment]
The project is awesome and we'd be keen to help with crowdfunding.

I'm personally very keen to help in any way I can, this has the potential to be the standard for LOB apps.

**Comment by grokys** [comment]
First of all, thanks for the enthusiasm everyone, it's fantastic to see! I've asked around a little and it seems that indeed [BountySource](https://www.bountysource.com/) or private agreements would probably be the best way to go for now. Maybe if this works out well we'd want to check out the [SF conservatory](https://sfconservancy.org/projects/services/)'s services.

The next step would be to find developers who want to be involved and try to formalize a process. A good first place to find developers would be our [gitter chat](https://gitter.im/AvaloniaUI/Avalonia) room I think.

In addition, as you mentioned @mrpastewart we don't have a decent website or docs currently. This is also something we need to work on (and it's a bit harder to find volunteers for this stuff!)

**Comment by webprofusion-chrisc** [comment]
Throwing in a different (and complimentary) approach from personal experience: forming a company and offering a pro/commerical version (annual subscription etc) that has a support/helpdesk ticket system behind it is a great way to get companies to pay you. It's sometimes difficult to convince a company to part with money to 'sponsor' something but if they can be demonstrably buying something (access to support). 

That doesn't mean you then spend all your time doing support it just means they can finally give you money to fix bugs/add features you were planning to do anyway. You then also have a corporate body that can pay other people to do things (devs, docs, design, support). If you don't make much money, you can't do much of those things. If you make lots of money, you can do lots of those things.

There are plenty of enterprise apps using WPF that would like the option to be cross platform (at least Mac OS and often iPads etc), so just give them the opportunity to pay you.

**Comment by VladimirAkopyan** [comment]
Guys, common, just take the money. 
It's been over a month. Setup project budget.

**Comment by grokys** [comment]
I've been advised by various people that https://opencollective.com/ is a good place to start.

**Comment by grokys** [comment]
I've applied to create an account on  https://opencollective.com, we'll see what happens from there I guess!

**Comment by grokys** [comment]
Here we go! https://opencollective.com/avalonia. Now to decide how to use any funds that come in!

**Comment by KvanTTT** [comment]
I think this link also should be placed to README :)

**Comment by grokys** [comment]
For anyone coming across this in future, we've now set up a company to provide paid support for Avalonia, and also paid contract work. Please get in touch with us if you're interested in either!

**Comment by kvalitetskontrollMarkaryd** [comment]
Up vote this and write a comment:

https://supportcenter.devexpress.com/ticket/details/t1014149/controls-for-avalonia-ui

**Comment by maxkatz6** [comment]
We have paid support officially 
https://github.com/AvaloniaUI/Avalonia/issues/4065
